webSites/webSites-war/src/main/java/org/silverpeas/components/websites/servlets/WebSitesRequestRouter.java

```diff
@@ -90,1056 +87,1137 @@ public WebSiteSessionController createComponentSessionController(
+      } else if (function.startsWith("FromWysiwyg")) {
+        destination = fromWysiwygEditor(scc, request, flag);
+      } else if ("TopicUp".equals(function)) {
+        destination = moveUpTopic(scc, request, flag);
+      } else if ("TopicDown".equals(function)) {
+        destination = moveDownTopic(scc, request, flag);
+      } else if ("PubUp".equals(function)) {
+        destination = moveUpPublication(scc, request, flag);
+      } else if ("PubDown".equals(function)) {
+        destination = moveDownPublication(scc, request, flag);
+      } else if (function.startsWith("modifDesc.jsp")) {
+        destination = modifySiteDescription(scc, request, flag);
+      } else if ("Suggest".equals(function)) {
+        request.setAttribute("AllIcons", scc.getAllIcons());
+        request.setAttribute("Action", "suggest");
+        destination = "/webSites/jsp/descBookmark.jsp";
+      } else if (function.startsWith("descBookmark.jsp")) {
+        request.setAttribute("AllIcons", scc.getAllIcons());
+        destination = "/webSites/jsp/descBookmark.jsp";
+      } else if (function.startsWith("descDesign.jsp")) {
+        request.setAttribute("AllIcons", scc.getAllIcons());
+        destination = "/webSites/jsp/descDesign.jsp";
+      } else if (function.startsWith("organize.jsp")) {
+        destination = organizeSites(scc, request, flag);
+      } else if ("AddTopic".equals(function)) {
+        destination = addTopic(scc, request, flag);
+      } else if (function.startsWith("updateTopic")) {
+        destination = updateTopic(scc, request, flag);
+      } else if (function.startsWith("classifyDeclassify.jsp")) {
+        destination = classification(scc, request, flag);
+      } else if (function.startsWith("manage.jsp")) {
+        destination = manageSite(scc, request, flag);
+      } else if (function.startsWith("design.jsp")) {
+        destination = designSite(scc, request, flag);
+      } else if (function.equals("EffectiveUploadFile")) {
+        destination = uploadFile(scc, request, flag);
+      } else if (function.startsWith("descUpload.jsp")) {
+        request.setAttribute("AllIcons", scc.getAllIcons());
+        destination = "/webSites/jsp/descUpload.jsp";
+      } else if (function.equals("EffectiveUploadSiteZip")) {
+        destination = uploadSite(scc, request, flag);
+      } else {
+        destination = "/webSites/jsp/" + function;
+      }
+    } catch (Exception e) {
+      request.setAttribute("javax.servlet.jsp.jspException", e);
+      destination = "/admin/jsp/errorpageMain.jsp";
+    }
 
-        if (action == null) {
-          id = "0";
-          action = "Search";
-        }
+    return destination;
+  }
 
-        FolderDetail webSitesCurrentFolder = scc.getFolder(id);
-        scc.setSessionTopic(webSitesCurrentFolder);
-        request.setAttribute("CurrentFolder", webSitesCurrentFolder);
+  private String uploadSite(final WebSiteSessionController scc, final HttpRequest request,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "uploadSite" is 12 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)